CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.7...3.30)
```

Comment:
What parts of your change would be incompatible with cmake 3? And if so: Isn't cmake 3.31 the latest of v3?

Reply:
Indeed, maybe make it 3.99 to be on the safe side (if 4 does introduce some backward incompatibilities)?

Reply:
This PR doesn't introduce any incompatible changes. The syntax is a bit confusing. The second argument to `cmake_minimum_required` sets the [minimum policy version](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#policy-version). It basically says: "this is a (policy-)version this project can build with". In CMake, this will set all policies introduced in that version to `NEW`. Since there aren't any policies used (AND CMake 4 doesn't warn about any implicitly used ones), there's no issue bumping this.

I can set it to 3.31.

Reply:
> This PR doesn't introduce any incompatible changes.

No, I meant CMake4's incompatibilities. The second argument of `cmake_minimum_required` is the *maximal* version that can be used with this project, isn't it? Then we'd need to update the build every time a new CMake-3 is released. Hence I suggest setting it to something that will never be overflown.

Reply:
> The second argument of `cmake_minimum_required` is the _maximal_ version that can be used with this project, isn't it?

Ah no, that's the confusion I meant. It's the ~~minimum~~ maximum policy version ([docs](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html)), so it only controls breaking changes _up to_ that version.

Reply:
Done.

Reply:
> Ah no, that's the confusion I meant. It's the ~minimum~ maximum policy version ([docs](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html)), so it only controls breaking changes _up to_ that version.

OK, I see - the last argument is about the _policy_ version, not cmake's version. Isn't it a sure thing to introduce incompatibilities? If someone uses the older cmake-3.7, it will behave differently from a newer version supporting some new policies introduced lately, right? I fail to see why it is desired - in general, and in the case of pdfpc specifically.

Perhaps this change should be removed from this PR.

Reply:
CMake will warn when a project relies on a policy that's set to `OLD`. Since that's not the case here, it's fine to bump the policy. In CMake 4, you get the following warning:

```
Compatibility with CMake < 3.10 will be removed from a future version of
CMake.

Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
to tell CMake that the project requires at least <min> but has been updated
to work with policies introduced by <max> or earlier.
```

But I agree, let's move this to another PR.